libs/execution_context/ebpf_maps.c

```diff
@@ -1063,6 +1085,80 @@ _update_queue_map_entry(
     return result;
 }
 
+static ebpf_core_map_t*
+_create_stack_map(_In_ const ebpf_map_definition_in_memory_t* map_definition)
+{
+    size_t stack_map_size =
+        EBPF_OFFSET_OF(ebpf_core_stack_map_t, slots) + map_definition->max_entries * sizeof(uint8_t*);
+    return _create_array_map_with_map_struct_size(stack_map_size, map_definition);
+}
+
+static void
+_delete_stack_map(_In_ ebpf_core_map_t* map)
+{
+    ebpf_core_stack_map_t* stack_map = EBPF_FROM_FIELD(ebpf_core_stack_map_t, core_map, map);
+
+    for (size_t i = 0; i < stack_map->core_map.ebpf_map_definition.max_entries; i++) {
+        ebpf_epoch_free(stack_map->slots[i]);
+    }
+    ebpf_epoch_free(stack_map);
+}
+
+static ebpf_result_t
+_find_stack_map_entry(
+    _In_ ebpf_core_map_t* map, _In_opt_ const uint8_t* key, _In_ bool delete_on_success, _Outptr_ uint8_t** data)
+{
+    if (!map || key)
+        return EBPF_INVALID_ARGUMENT;
+
+    ebpf_core_stack_map_t* stack_map = EBPF_FROM_FIELD(ebpf_core_stack_map_t, core_map, map);
+    ebpf_lock_state_t state = ebpf_lock_lock(&stack_map->core_map.lock);
+    if (stack_map->top_of_stack == 0) {
+        *data = NULL;
+    } else {
+        *data = stack_map->slots[stack_map->top_of_stack - 1];
+        if (delete_on_success) {
+            ebpf_epoch_free(stack_map->slots[stack_map->top_of_stack - 1]);
+            stack_map->slots[stack_map->top_of_stack - 1] = NULL;
+            stack_map->top_of_stack--;
+        }
+    }
+    ebpf_lock_unlock(&stack_map->core_map.lock, state);
+    return *data == NULL ? EBPF_OBJECT_NOT_FOUND : EBPF_SUCCESS;
+}
+
+static ebpf_result_t
+_update_stack_map_entry(
+    _In_ ebpf_core_map_t* map, _In_ const uint8_t* key, _In_opt_ const uint8_t* data, ebpf_map_option_t option)
+{
+    ebpf_result_t result;
+    if (!map || key || (option == EBPF_NOEXIST))
+        return EBPF_INVALID_ARGUMENT;
+
+    ebpf_core_stack_map_t* stack_map = EBPF_FROM_FIELD(ebpf_core_stack_map_t, core_map, map);
+    ebpf_lock_state_t state = ebpf_lock_lock(&stack_map->core_map.lock);
+
+    // Check for stack full.
+    if (stack_map->top_of_stack >= stack_map->core_map.ebpf_map_definition.max_entries) {
+        result = EBPF_OUT_OF_SPACE;
+        goto Done;
+    }
+
+    uint8_t* value = ebpf_epoch_allocate(stack_map->core_map.ebpf_map_definition.value_size);
+    if (!value) {
+        result = EBPF_NO_MEMORY;
+        goto Done;
+    }
+    memcpy(value, data, stack_map->core_map.ebpf_map_definition.value_size);
```

Comment:
I wonder why SAL is not complaining as data can be NULL. I understand stack maps cant delete by update only find_and_delete is allowed - and hence data must be !NULL in update. But still , shouldn't we check?